src/scene_actortarget.cpp

```diff
@@ -121,7 +121,9 @@ void Scene_ActorTarget::UpdateSkill() {
 			return;
 		}
 		if (Main_Data::game_party->UseSkill(id, actor, target_window->GetActor())) {
-			Game_System::SePlay(Game_System::GetSystemSE(Game_System::SFX_UseItem));
+			const RPG::Sound* sound = Game_System::GetFirstSound(ReaderUtil::GetElement(Data::animations, ReaderUtil::GetElement(Data::skills, id)->animation_id));
+			if (sound != nullptr)
+				Game_System::SePlay(*sound);
```

Comment:
This needs sanity checks because this is external data which is untrusted.
Look at other places where ReaderUtil::GetElement is used and add similiar Output::Warning handling.
At least for the animation because the skill is already sanitized in UseSkill()

Reply:
GetFirstSound() already checks if the animation is "nullptr" and returns a "nullptr" sound in that case. But if you still prefer that I make a sanity check for animation, then I'll do it.

Reply:
yes pls do a check for a valid animation beforehand imo functions where NULL is just an invalid argument the argument type should be reference & and not pointer \*.